### ISSUE TITLE
jazzPanda

### ISSUE THREAD

**Issue by MelodyJIN-Y** [Bioconductor/Contributions#3538]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/phipsonlab/jazzPanda

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand Bioconductor [Package Naming Policy][9] and acknowledge
  Bioconductor may retain use of package name.

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [x] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website][8] to sign up.

[1]: https://contributions.bioconductor.org/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/
[8]: https://bioconductor.org/
[9]: https://bioconductor.org/developers/package-submission/#naming

**Comment by lshep** [comment]
Packages are generally expected to use standard Bioconductor class objects.  Please expand to use standard class objects like SpatialExperiment or SingleCellExperiment.  For What Its Worth you might also check out the  Experiment Data Package  [TENxXeniumData](https://bioconductor.org/packages/release/data/experiment/html/TENxXeniumData.html) that already provides Xenium data in such a standard class object.

**Comment by lshep** [comment]
May we expect updates soon?

**Comment by MelodyJIN-Y** [comment]
Hi, I am  overseas and will make updates in 2 weeks. Apologies for the delay.

**Comment by MelodyJIN-Y** [comment]
Hi, 
Thank you for your comments, and I apologize for the delay in updates.

* We have introduced a new function, convert_data, which converts data from the SingleCellExperiment, SpatialExperiment, and SpatialFeatureExperiment classes into a simple list format that can be directly passed to jazzPanda. We have included some examples in the vignette to aid with this new feature. We will continue to add support for other data classes as new ones emerge in the field of spatial data.

* We appreciate your suggestion regarding the TENxXeniumData/ExperimentHub package. Upon review, we found that the available datasets from TENxXeniumData and ExperimentHub do not include multi-sample examples that would demonstrate the full capabilities of our package.

**Comment by lshep** [comment]
Your package should take a SingleCellExperiment/SpatialExperiment class directly;  it should not be flattened to a list and then used to create a Seurat object. Seurat objects are not valid Bioconductor class objects.   You certainly can keep the interaction with Seurat but you should also make the package directly accept the standard Bioconductor class object (and ideally be endomorphic to return the same class object as output as input)

**Comment by MelodyJIN-Y** [comment]
Hi, 

Thank you for your feedback. We have updated our package to accept SingleCellExperiment/SpatialExperiment/SpatialFeatureExperiment classes directly as input. This change aligns with Bioconductor standards and should improve user experience

**Comment by lshep** [comment]
I apologize for the delay, I was away on holiday. I should have a review today or tomorrow at the latest.

**Comment by lshep** [comment]
Thank you for the submission and I do apologize for the delays.  Please see the
following: 



**README**

- [ ] Update Installation to using BiocManager in README

**DESCRIPTION**

- [ ] License should just be GPL-3 there is no additional file


**vignette**

- [ ] The code that actually uses Bioconductor classes and executes your package
  functions are all eval=FALSE.  The code chunks should be evaluted.  

- [ ] Most people will not have direct access to or know where to find
  `inst/generate_vignette_data.R` the following will point to the script on
  their system `system.file("script","generate_vignette_data.R",
  package="jazzPanda")`

- [ ] The From SpatialExperiment object converts the object to Seurat but then
  in the From a seurat object section it converts to a SingleCellExperiment.
  This seems very odd and that each could maintain their sturcutre? 

- [ ] If all the clustering for the objects is the same, it might make sense to
  show all the data loading first, then the clustering steps. That also being
  said, You should be able to do clustering off Bioconductor object not having
  to convert to a Seurat object. Either you need to implement the functions you
  have to work on the Bioconductor objects or look for the equivalent
  in Bioconductor that is being performed by Seurat.
  
- [ ] It would be nice to see the visualization used with the Bioconductor
  object examples instead of the random data frames. because you seperate the
  loading of that data then show all the obejcts and then go back to using them
  it gets disjointed and hard to follow.

- [ ] The clustering and plotting is a lot to digest. It might be helpful to add
  more clarifying comments and text to describe to the user what is going
  on. Its hard for end users to understand what large big chunks do.

- [ ] have a distinct section for sessionInfo

**man**

- [ ] Have a package level man page so a naive user that does ?jazzPanda will
  get some starting information

**R code**

- [ ] Remove commented out code. Comments to describe whats going on is
  encouraged but if code is not needed it should not be there.

- [ ] Internal functions could be grouped and prefaced with a `.` to indicate
  they are not exported. 

- [ ] So get_lasso_coef does cannot take a Bioconductor object. The previous
  functions take in the Bioconductor objects and then return lists? I guess this
  is true for all your functions. Is it not possible to keep the class
  structures? By not having endomorphic functions it  makes it difficult to be
  used as part of workflow steps. If its matrix by sampels or genes, shouldn't
  it be able to be added as additional matricies within the strucutres? 

- [ ] If not it might also be good to define an output class and have defined
  accessors esp if they are highly used in your example plotting. 


When ready please push to git.bioconductor.org to trigger a new build and
comment on the above observations. We look forward to exediating this process
moving forward and including your package into Bioconductor. 

Cheers,

**Comment by MelodyJIN-Y** [comment]
Thanks for your feedback.
I've made the following changes based on your comments:

README


> - [x] Update Installation to using BiocManager in README


DESCRIPTION

> - [x] License should just be GPL-3 there is no additional file

vignette

> - [x] The code that actually uses Bioconductor classes and executes your package
> functions are all eval=FALSE. The code chunks should be evaluted.

> - [x] Most people will not have direct access to or know where to find
> inst/generate_vignette_data.R the following will point to the script on
> their system system.file("script","generate_vignette_data.R", package="jazzPanda")


> - [ ] The From SpatialExperiment object converts the object to Seurat but then
> in the From a seurat object section it converts to a SingleCellExperiment.
> This seems very odd and that each could maintain their sturcutre?


  - *The sections you've mentioned (From a SpatialExperiment/ SpatialFeatureExperiment/SingleCellExperiment/ Seurat object) are indeed intended as examples to demonstrate how users can use the function with data in various object types. The use of conversion functions in these examples was simply to create the corresponding objects that users may have. Each section aims to provide a guidance for users who may start with one type of object and need to know how the function behaves with other types as well.*
  - *We have updated the vignette to improve clarity*



> - [ ] If all the clustering for the objects is the same, it might make sense to
> show all the data loading first, then the clustering steps. That also being
> said, You should be able to do clustering off Bioconductor object not having
> to convert to a Seurat object. Either you need to implement the functions you
> have to work on the Bioconductor objects or look for the equivalent
> in Bioconductor that is being performed by Seurat.
> 

- *The clustering section in our documentation serves as a simple example for illustrative purposes. Our package is designed with the assumption that users have already performed clustering using their preferred methods and have the clustering labels ready for use.*
- *For better integration with Bioconductor, we have changed our example clustering to use scran based on the suggestions from the "Principles of Spatial Transcriptomics Analysis with Bioconductor"*



> - [ ] It would be nice to see the visualization used with the Bioconductor
> object examples instead of the random data frames. because you seperate the
> loading of that data then show all the obejcts and then go back to using them
> it gets disjointed and hard to follow.


- *The visualization is based on actual data from a SpatialExperiment object, rather than random data frames.*



> - [x] The clustering and plotting is a lot to digest. It might be helpful to add
> more clarifying comments and text to describe to the user what is going
> on. Its hard for end users to understand what large big chunks do.


> - [x] have a distinct section for sessionInfo


man

> - [x] Have a package level man page so a naive user that does ?jazzPanda will
> get some starting information

R code

> - [x] Remove commented out code. Comments to describe whats going on is
> encouraged but if code is not needed it should not be there.
> 


> - [x] Internal functions could be grouped and prefaced with a . to indicate
> they are not exported.


> - [x] So get_lasso_coef does cannot take a Bioconductor object. The previous
> functions take in the Bioconductor objects and then return lists? I guess this
> is true for all your functions. Is it not possible to keep the class
> structures? By not having endomorphic functions it makes it difficult to be
> used as part of workflow steps. If its matrix by sampels or genes, shouldn't
> it be able to be added as additional matricies within the strucutres?

- *The get_lasso_coef function is designed for internal use only. Additionally, we have introduced new class structures to enhance clarity.*




> - [x] If not it might also be good to define an output class and have defined
> accessors esp if they are highly used in your example plotting.
> 

Thanks
Melody

**Comment by lshep** [comment]
The default branch of your GitHub repository has been added to Bioconductor's
git repository as branch devel.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/MelodyJIN-Y.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("jazzPanda")`. The package 'landing page' will be created at

https://bioconductor.org/packages/jazzPanda

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/
